VoodooPS2Keyboard/VoodooPS2Keyboard.cpp

```diff
@@ -336,6 +347,41 @@ ApplePS2Keyboard* ApplePS2Keyboard::probe(IOService * provider, SInt32 * score)
 
 // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 
+IORegistryEntry* ApplePS2Keyboard::getDisplay() {
+    IORegistryEntry* entry = nullptr;
+    IORegistryEntry* display = nullptr;
+
+    // Waiting for IGPU
+    size_t counter = 20;
+    while (counter--) {
+        if ((entry = IORegistryEntry::fromPath("/PCI0/IGPU", gIODTPlane)))
```

Comment:
Ugh, why not to iterate over the devices and look for an intel-based vendor. This is just ugly, as not all PCI root hubs are named PCI0 even. Have a look at https://github.com/acidanthera/Lilu/blob/master/Lilu/Sources/kern_devinfo.cpp

Reply:
Sorry, I have little knowledge with this situation. I have also found some [gen1 laptop examples](https://www.tonymacx86.com/threads/mojave-on-gen1-laptop-problems.262967/) but the IGPU is still under `PCI0`. 
My assumption here is that the user either had the old-school `GFX0` to `IGPU` patch, or adopted WhateverGreen that provided gen1 support recently.

Reply:
Like I said, it is not always the case. Especially on devices with multiple PCI lanes and exotic chipsets. Please change that.

Reply:
I felt exhausted finding a dynamic path to IGPU for the past two day. I have rebooted about two hundred times, while my laptop is not resilient and got mysterious behavior like missing WLAN or boot entry following KPs. And recently my life's getting busier, maybe I won't have much time to test one case repeatedly for the coming month.

For the case of laptop DSDT, I assume OEM need to follow Intel's reference code tightly unlike desktop, and so far I haven't seen an exception. Something I learned before is that, when an OEM want to get direct CPU connection for TB controller instead of south bridge, Intel have threatened to stop further support. So I suppose they also won't rename `PCI0` for no good.

And for implementation, I have consider whether to add Lilu as a dependency. However, VoodooPS2 is a quite fundamental kernel extension that requires minimal dependency. Also when Lilu present, weg would have renamed the device as well. So I think `/PCI0/IGPU` is just kind of implicit requirement.

I extracted the method to find child device by address, and thought maybe then I could find the display from root to /PCI0@**0**/IGPU@**2**/DD1F@**400**. (0:2:400?) But it just panics when I changed the starting point to parent device of IGPU. Also, `OSDynamicCast` seems broken when I want to replace the type conversion for device. Wasted two hours to find that out since I thought it was safe as I used it before.